### PR TITLE
3.0 Introduce Authentication and User Management BDDs

### PR DESCRIPTION
**Pull request by farost**
## Usage and product changes
We introduce behaviour steps declarations for authentication and user management. The test is relevant only for connecting drivers, so it has been moved to the `driver` package.

Additionally, `insert` and `delete` feature files has been updated to unlock more steps for usage against 3.0 (refactored query execution, session-transaction management, and some queries). Note that these files are still not fully suitable and require more refactoring and testing.

### CODE REVIEW

File: concept/thing/attribute.feature [hunk 1]
```diff
@@ -43,7 +43,7 @@ Feature: Concept Attribute
     Given attribute(email) set annotation: @regex("\S+@\S+\.\S+")
     Given transaction commits
     Given connection open write transaction for database: typedb
-    Given set time zone: Europe/London
+    Given set time-zone: Europe/London
```
Comment (farost) [inline concept/thing/attribute.feature:46]: We had two different types of these steps, I inclined to this one following [this](https://english.stackexchange.com/questions/3934/time-zone-vs-timezone)

Will require server's update when the ref is updated on the server side! Don't be scared

File: query/language/insert.feature [hunk 2]
```diff
@@ -520,96 +514,6 @@ get;
       | key:ref:0 |
 
 
-  Scenario: when an attribute owns an attribute, an instance of that attribute can be inserted onto it
```
Comment (farost) [inline query/language/insert.feature:523]: Not relevant anymore

File: query/language/delete.feature [hunk 3]
```diff
@@ -223,42 +185,8 @@ Feature: TypeQL Delete Query
     Then answer size is: 0
 
 
-  Scenario: an instance can be deleted using its own type label
```
Comment (farost) [inline query/language/delete.feature:226]: Not relevant? We just delete it through `delete $x;`

File: query/language/delete.feature [hunk 4]
```diff
@@ -1047,153 +957,101 @@ Feature: TypeQL Delete Query
...
+    Then typeql write query; fails
       """
       match
         $x isa person, has address $a;
       delete
-        $x has thing $a;
-      """
-
-
-  Scenario: an attribute can be specified by direct type when deleting an ownership of it
```
Comment (farost) [inline query/language/delete.feature:1113]: Not relevant anymore

File: driver/user.feature [hunk 5]
```diff
@@ -0,0 +1,472 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# TODO: remove ignores once bootup can be configured through BDD
+
+# TODO: Change expected error messages when implemented
```
Comment (farost) [inline driver/user.feature:7]: I spammed "Invalid credential supplied" everywhere for now

File: driver/user.feature [hunk 6]
```diff
@@ -0,0 +1,472 @@
...
+    Given typedb starts
+    Given connection opens with username 'admin', password 'password'
+    When create user with username 'user', password 'password'
+    When delete user: user
+    Then delete user: user
+    Then delete user: user2
+    Then delete user: surely-non-existing-user
+
+
+    # TODO: Not sure if it's correct, may be implemented differently
```
Comment (farost) [inline driver/user.feature:177]: Not sure

File: driver/user.feature [hunk 7]
```diff
@@ -0,0 +1,472 @@
...
+    Given typedb starts
+    Given connection opens with username 'admin', password 'password'
+    When create user with username 'user', password 'password'
+    Then get current username: admin
+    When connection closes
+    When connection opens with username 'user', password 'password'
+    Then get current username: user
+
+  ##################
+  # CONFIG OPTIONS #
```
Comment (farost) [inline driver/user.feature:339]: Rewritten the steps, left commented out as we don't have these options

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**